### DISCUSSION TITLE
Multiple record query to the same collection fail

### DISCUSSION THREAD

**Discussion by Inferis** [pocketbase/pocketbase#622] [Q&A]
I've got a small app that due to how it's set up does multiple similar requests/queries (different request params) on the same collection. When I execute then, the first ones fail, and only the last one correctly finishes returning the expected data. 

Is this to be expected (I certainly didn't)?

I can probably make a standalone small project/site that shows this (haven't tried yet).

So, to be clear:

"await" Query 1 - query foo { filter = baz, index = 1 }
"await" Query 2 - query foo { filter = baz, index = 10 }
"await" Query 3 - query foo { filter = baz, index = 2 }

These are executed in rapid succession since awaiting will yield to another continuation. Query 1 and Query 2 fail, and Query 3 finishes with the expected data.

**Comment by ganigeorgiev** [comment]
If you are using the JS SDK, this is probably related to the auto cancellation - https://github.com/pocketbase/js-sdk#auto-cancellation.

To disable it, you can set either `$autoCancel: false` or define a unique `$cancelKey` for each request:

```js
client.records.getList('example', 1, 20, { '$autoCancel': false });
client.records.getList('example', 1, 20, { '$autoCancel': false });
client.records.getList('example', 1, 20, { '$autoCancel': false });
```

_I'll move the issue to Q&A discussion._
